### ISSUE TITLE
Nested Filtering in Query Builder

### ISSUE THREAD

**Issue by robarivas** [DaxStudio/DaxStudio#839]
Query Builder in DAX Studio allows "AND" logic between/across columns/fields while also allowing "OR" logic within columns/fields. It would be great, however, if nested filtering could be easily achieved in the GUI of the Query Builder feature. For example:

field1 = value_x
and field 2 = value_y
and (
         or ( field 3 in ( value_a, value_b, value_c ), field 4 in ( value_d, value_e )
         )
)

If a user wants to achieve such nested filtering logic they must edit the DAX code by hand.

**Comment by dgosbell** [comment]
> It would be great, however, if nested filtering could be easily achieved in the GUI of the Query Builder feature

Yes, it would be great, but I have not been able to think of a design which allows this sort of flexibility, but does not add too much complexity.  I want to keep the UI simple and intuitive. I think at the moment you can build 90% of the common queries and for the remaining 10% you can build out the core query and then just generate the text and edit it by hand. 

It also gets much more complex if each of the fields in your example came from different tables. So not only would the UI get more complex - I would also have to generate a much more complex filter expressions.
